initrd/kraftignore.go

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2022, Unikraft GmbH and The KraftKit Authors.
+// Licensed under the BSD-3-Clause License (the "License").
+// You may not use this file except in compliance with the License.
+package initrd
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"kraftkit.sh/log"
+)
+
+// kraftignore filename
+const KraftignoreFileName = ".kraftignore"
+
+type IgnoringFileType string
+
+const (
+	Exist    = IgnoringFileType("Exist")
+	NotExist = IgnoringFileType("NotExist")
+	SkipDir  = IgnoringFileType("SkipDir")
+)
+
+// returns file and directory names specified in .kraftignore
+func getKraftignoreItems(ctx context.Context, initrd *directory) ([]string, error) {
```

Comment:
Let's expose this as a public method that accepts an input directory. This allows the method to be used for other applications outside of the `*directory` implementation.

```diff
- // returns file and directory names specified in .kraftignore
- func getKraftignoreItems(ctx context.Context, initrd *directory) ([]string, error) {
+ // GetKraftIgnoreItems returns file and directory names specified in .kraftignore
+ func GetKraftIgnoreItems(ctx context.Context, dir string) ([]string, error) {
```
You then simply need to replace `initrd.opts.kraftignorePath` with `dir`.